packages/roosterjs-editor-plugins/lib/PasteManager/PasteManager.ts

```diff
@@ -94,25 +102,70 @@ export default class PasteManager implements EditorPlugin {
     };
 }
 
-function getOnPasteHandler(): OnPasteHandler {
-    if (browserData.isChrome || browserData.isFirefox || browserData.isEdge) {
-        return onPaste;
-    } else if (browserData.isIE) {
-        return onPasteIE;
-    } else if (browserData.isSafari) {
-        return onPasteSafari;
+function getClipboardData(dataTransfer: DataTransfer): ClipboardData {
+    let clipboardData: ClipboardData = {
+        imageData: {},
+        textData: dataTransfer.getData('text'),
+        htmlData: null,
+    };
+
+    let image = getImage(dataTransfer);
+    if (image) {
+        clipboardData.imageData.file = image;
+    }
+
+    return clipboardData;
+}
+
+function getImage(dataTransfer: DataTransfer): File {
+    // Chrome, Firefox, Edge support dataTransfer.items
+    let fileCount = dataTransfer.items ? dataTransfer.items.length : 0;
+    for (let i = 0; i < fileCount; i++) {
+        let item = dataTransfer.items[i];
+        if (item.type && item.type.indexOf('image/') == 0) {
+            return item.getAsFile();
+        }
+    }
+    // IE, Safari suuport dataTransfer.files
+    fileCount = dataTransfer.files ? dataTransfer.files.length : 0;
+    for (let i = 0; i < fileCount; i++) {
+        let file = dataTransfer.files.item(i);
+        if (file.type && file.type.indexOf('image/') == 0) {
+            return file;
+        }
+    }
+    return null;
+}
+
+function getHtml(editor: Editor, callback: (container: HTMLElement) => void) {
+    // cache original selection range in editor
+    let originalSelectionRange = editor.getSelectionRange();
+
+    if (!pasteContainer) {
+        pasteContainer = fromHtml(CONTAINER_HTML)[0] as HTMLElement;
+        document.body.appendChild(pasteContainer);
     } else {
-        // let browser handle paste itself if it's not a known browser
-        return null;
+        pasteContainer.style.display = '';
```

Comment:
It is null when first time call this function. Then it will be reused.
Setting display to '' makes the div to be shown, same effect with 'block' since 'block' is the default value for a DIV.

Reply:
I see. But I suppose this container is just for storing the paste content, we don't want to display that, why not always set it to hidden?

Reply:
it is required to show the container so that we can put in focus and accept the pasted content. A hidden element can't be focused.

Reply:
Got it.